portfolio/src/App.jsx

```diff
@@ -1,157 +1,145 @@
-              <a href="https://tfkjosh.github.io/Password-generator/" target="_blank" class="project 3"></a>
-              <img src={Project3} alt="https://tfkjosh.github.io/Password-generator/" className="w-full h-auto" />
-              <h3 className="text-2xl font-semibold mt-8">
-              This application is designed for an employee to generate a random password based on criteria they've selected.
-              </h3>
-              <p className="text-gray-400 text-sm mt-4">https://github.com/tfkjosh/Password-generator</p>
-              <div className="flex justify-evenly mt-12">
-                <button className="px-6 py-3 text-lg bg-gradient-to-t from-orange-500 rounded-full to-yellow-500 hover:from-orange-700 hover:to-yellow-700">
-                  Live preview
-                </button>
-                <button className="px-6 py-3 border rounded-full hover:border-orange-500 hover:text-yellow-500">
-                  Checkout github
-                </button>
-              </div>
-            </div>
-            <div className="border border-gray-500 rounded-md p-5">
-              {/*https://pwa-challenge1-b508091ee337.herokuapp.com/*/}
-              <a href="https://pwa-challenge1-b508091ee337.herokuapp.com/" target="_blank" class="project 4"></a>
-              <img src={Project4} alt="https://pwa-challenge1-b508091ee337.herokuapp.com/" />
-              <h3 className="text-2xl font-semibold mt-8">
-              This application features a number of data persistence techniques, insuring the application runs regardless of browser supported function. 
-              </h3>
-              <p className="text-gray-400 text-sm mt-4">https://github.com/tfkjosh/PWA-Challenge</p>
-              <div className="flex justify-evenly mt-12">
-                <button className="px-6 py-3 text-lg bg-gradient-to-t from-orange-500 rounded-full to-yellow-500 hover:from-orange-700 hover:to-yellow-700">
-                  Live preview
-                </button>
-                <button className="px-6 py-3 border rounded-full hover:border-orange-500 hover:text-yellow-500">
-                  Checkout github
-                </button>
-                </div>
-              </div>
+            <div className="flex flex-col sm:flex-col gap-12 mt-11">
+              {projects.map(
+                ({ imgSrc, description, githubHref, livePreviewHref }) => (
+                  <div className="border border-gray-500 rounded-md">
+                    <a href={githubHref} target="_blank">
+                      <img
+                        src={imgSrc}
+                        alt={githubHref}
+                        className="w-full h-auto"
+                      />
+                    </a>
+                    <div className="p-5">
+                      <h3 className="text-2xl font-semibold">{description}</h3>
+                      <div className="flex gap-4 justify-end mt-8">
+                        {livePreviewHref && (
```

Comment:
This is some fancy shorthand syntax which can be read as the following:

If `livePreviewHref` is not undefined (if it exists for the given list/array element), then add the following JSX element to the UI. 

Given that not all of your projects have been hosted live (at least, I figured that was the case given that all but one of the Live Preview buttons didn't have any URL to navigate the user to), I figured it would make sense to only show this button whenever it actually works by directing the user to a live preview. If no live preview exists yet, it might make sense to leave this button out, in those cases.